compiler/parser/src/lexer.rs

```diff
@@ -439,10 +438,9 @@ where
     fn take_number(&mut self, radix: u32) -> Option<char> {
         let take_char = Lexer::<T>::is_digit_of_radix(self.window[0], radix);
 
-        if take_char {
-            Some(self.next_char().unwrap())
-        } else {
-            None
+        match take_char {
+            true => Some(self.next_char().unwrap()),
+            _ => None,
```

Comment:
I like the idea of pattern matching, but if expression should be preferred to match expression when we are dealing with a boolean value.

Reply:
I'll restore if expression in `take_number`

Reply:
note that this could probably use [`bool.then`](https://doc.rust-lang.org/std/primitive.bool.html#method.then)

Reply:
Thank you for the adivse.